integration-tests/jiva/replicadeletion/jiva_replica_deletion_test.go

```diff
@@ -0,0 +1,181 @@
+		defaultReplicaLabel = "openebs.io/replica=jiva-replica"
+		// defaultCtrlLabel represents the jiva controller
+		defaultCtrlLabel = "openebs.io/controller=jiva-controller"
+		// replicaLabel consist of defaultReplicaLabel and coressponding
+		// pvcLabel
+		replicaLabel string
+		// ctrlLabel consist of defaultReplicaLabel and coressponding
+		// pvcLabel
+		ctrlLabel string
+	)
+	BeforeEach(func() {
+
+		//Extracting storageclass artifacts unstructured
+		jivaSCUnstructured, err := artifacts.GetArtifactUnstructured(
+			artifacts.Artifact(jivaSCYaml),
+		)
+		Expect(err).ShouldNot(HaveOccurred())
+
+		//Extracting PVC artifacts unstructured
+		pvcUnstructured, err := artifacts.GetArtifactUnstructured(
+			artifacts.Artifact(pvcYaml),
+		)
+		Expect(err).ShouldNot(HaveOccurred())
+
+		//Apply jiva storageClass
+		cu := k8s.CreateOrUpdate(
+			k8s.GroupVersionResourceFromGVK(jivaSCUnstructured),
+			jivaSCUnstructured.GetNamespace(),
+		)
+
+		By("Deploying jiva-1r storageClass")
+		_, err = cu.Apply(jivaSCUnstructured)
+		Expect(err).ShouldNot(HaveOccurred())
+
+		//Apply PVC using jiva storageClass in jiva-test namespace
+		cu = k8s.CreateOrUpdate(
+			k8s.GroupVersionResourceFromGVK(pvcUnstructured),
+			pvcUnstructured.GetNamespace(),
+		)
+
+		By("Deploying PVC using jiva-1r storageClass in jiva-test namespace")
+		_, err = cu.Apply(pvcUnstructured)
+		Expect(err).ShouldNot(HaveOccurred())
+
+		// pvcLabel represents the coressponding pvc
+		pvcLabel := "openebs.io/persistent-volume-claim=" + pvcUnstructured.GetName()
+		replicaLabel = defaultReplicaLabel + "," + pvcLabel
+		ctrlLabel = defaultCtrlLabel + "," + pvcLabel
+
+		// Verify creation of jiva ctrl pod
+		_ = checkPodUpandRunning(string(jivaTestNamespace), ctrlLabel, 1)
+
+		// Verify creation of jiva replica pod
+		_ = checkPodUpandRunning(string(jivaTestNamespace), replicaLabel, 1)
+	})
+
+	AfterEach(func() {
+		//Extracting PVC artifacts unstructured
+		pvcUnstructured, err := artifacts.GetArtifactUnstructured(
+			artifacts.Artifact(pvcYaml),
+		)
+		Expect(err).ShouldNot(HaveOccurred())
+
+		//Extracting storageclass artifacts unstructured
+		jivaSCUnstructured, err := artifacts.GetArtifactUnstructured(
+			artifacts.Artifact(jivaSCYaml),
+		)
+		Expect(err).ShouldNot(HaveOccurred())
+
+		//Delete the PVC using it's unstructured format
+		cu := k8s.DeleteResource(
+			k8s.GroupVersionResourceFromGVK(pvcUnstructured),
+			string(jivaTestNamespace),
+		)
+		By("Deleting PVC in jiva-test namespace")
+		err = cu.Delete(pvcUnstructured)
+		Expect(err).ShouldNot(HaveOccurred())
+
+		//Delete jiva1-r storageClass
+		cu = k8s.DeleteResource(
+			k8s.GroupVersionResourceFromGVK(jivaSCUnstructured),
+			jivaSCUnstructured.GetNamespace(),
+		)
+		By("Deleting jiva-1r storageClass")
+		err = cu.Delete(jivaSCUnstructured)
+		Expect(err).ShouldNot(HaveOccurred())
+	})
+
+	Context("jiva replica deletion test", func() {
+		It("Deletion of jiva replica pod", func() {
```

Comment:
It() is not required here. Can the format should be:
```go
Context("node stickiness with jiva replica pod deletion", func() {
  By("fetching node name of jiva replica pod")
  // logic to set node name of jiva replica pod
  
  By("starting a loop to delete jiva replica pod")
  // logic
  By("deleting the running jiva pod")
  // logic
  By("waiting till jiva pod starts running")
  // logic
  By("verifying pod node matches with its old instance")
  // logic
})
```

Reply:
Without It() we can't execute the test. It() is necessary to run the test

Reply:
@mittachaitu the `// logic` implies `It()` calls
I want the code to be readable above all things.

Reply:
Done with latest commit